gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -231,6 +231,7 @@ var gpii = fluid.registerNamespace("gpii");
         } else {
             // we cant simply use fluid.filterKeys because that wont handle the cases where
             // there are 'undefined' values for the keys in handlerSpec.settings
+            // TODO: This is the wrong place for this filtering - it should have already happened in the MatchMaker
```

Comment:
I might be misremembering, but I believe the reason for delaying the filtering until this point, is that the transformation of common terms into application specific settings does not happen until after the matchmaker. This means that we dont have the (full) list of application settings to filter until this point...